src/test/groovy/pl/cleankod/util/MoneyRoundingUtilSpecification.groovy

```diff
@@ -0,0 +1,40 @@
+package pl.cleankod.util;
+
+import spock.lang.Specification
+
+import java.math.RoundingMode;
+
+class MoneyRoundingUtilSpecification extends Specification {
+
+    def "should provide expected values after rounding"() {
+        when:
+        def actualObject = MoneyRoundingUtil.roundHalfEven(givenValue)
+
+        then:
+        actualObject == expectedValue
+
+        where:
+        givenValue << [new BigDecimal(0.004501d).setScale(6, RoundingMode.HALF_UP),
+                       new BigDecimal(0.005001d).setScale(6, RoundingMode.HALF_UP),
+                       new BigDecimal(0.005401d).setScale(6, RoundingMode.HALF_UP),
+                       new BigDecimal(0.005501d).setScale(6, RoundingMode.HALF_UP),
+                       new BigDecimal(0.005601d).setScale(6, RoundingMode.HALF_UP)]
+        expectedValue << [new BigDecimal("0.00"),
```

Comment:
This should have given you a clue about a potential problem with the rounding assumptions.

Reply:
if you look at the code, it will pretty much give you a clear overview how I made rounding, in 2 phases of rouding